src/main/webapp/js/common/react/Navbar.js

```diff
@@ -55,5 +54,10 @@ function NavbarLink(props) {
   );
 }
 
+NavbarLink.propTypes = {
```

Comment:
Could we define this in prop_types? or maybe define the other prop_types where they are used. I'm not completely sure what is happening but, to me it looks like you are defining types to avoid errors with user input. If I am right, I think that makes a lot sense. having a defined way to do this that doesn't vary with the file might be beneficial.

Reply:
Indirectly this is somewhat validating user input, however the main feature of Prop Types is validating the [props](https://www.w3schools.com/react/react_props.asp) of a component to make sure that everything needed for the component to function is being inputted and everything is of the correct type. 

As for defining this in `prop_types.js`, I see what you're saying, but I think it'd be best if we followed the React conventions [which you can see on the React page on Prop Types](https://reactjs.org/docs/typechecking-with-proptypes.html). This includes setting the `propTypes` from within the file that defines the component.

Reply:
That makes sense! I saw that the project preview prop types were done differently, is there a reason for this?

Reply:
Ahh good catch. So to better explain what I'm thinking, I'd like to keep `propType` assignments within each component class. The reason that the Project components are different is because there is a lot of repeat there with prop types as data is sent down from parent to child components. So to follow DRY, I added the repeating prop types to a separate file that each imports individually.